plugins/nanospell/plugin.js

```diff
@@ -809,6 +808,7 @@
 					if (!uniqueWords[word] && self.validWordToken(word) && (typeof(spellcache[word]) === 'undefined')) {
 						words.push(word);
 						uniqueWords[word] = true;
+						spellcache[word] = true; // mark the word as correct, when the RPC returns, it will set this properly.
```

Comment:
This causes a word to not be put into an HTTP request if it's already been requested.  There is a chance that this causes a word to be marked correct until you start typing into the paragraph, but I suspect this happens less in real life + doesn't actually matter as much.  this does, however, significantly reduce the number of http calls made because common real words will not be included in the batch of 200, more than once.